sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -108,6 +108,7 @@ object Cast {
    */
   def needsTimeZone(from: DataType, to: DataType): Boolean = (from, to) match {
     case (StringType, TimestampType | DateType) => true
+    case (DateType, StringType) => true
```

Comment:
It could be symmetric to line above:
```scala
case (TimestampType | DateType, StringType) => true
``` 
So we could replace:
```scala
case (TimestampType, StringType) => true
```

Reply:
can you do a followup to clean this up? thanks!

Reply:
Here is it: https://github.com/apache/spark/pull/30223